### PR TITLE
Feature/navigation

### PR DESCRIPTION
**Pull request by ericahagle**
# Description
- Adds `EncounterBuilderButton` to header/nav
- Adds `LogoutButton` to header/nav
- Routes `EncounterBuilderButton` to `EncounterBuilder`
- Routes `LogoutButton` to `Login`
- Adds dark-mode-first styling to header/nav
- Adds responsive styling to header/nav

# Screenshot(s)
Screenshot 2024-02-04 at 10 26 48 PM
Screenshot 2024-02-04 at 10 27 11 PM
Screenshot 2024-02-04 at 10 27 02 PM

# Notes
- Colors and fonts are 100% negotiable, but I wanted to take a stab at some styling (because I really need the css practice, mostly! 😂 )
- We might want to add some conditional logic to show/hide the buttons on different pages, but I figured that could wait until we actually have the pages all set up

# Checklist
- [x] My PR has an appropriately descriptive and concise title.
- [x] I ran the code locally and verified that there are no visible errors.
- [x] feat: My PR clearly describes the feature I'm adding.

### CODE REVIEW

File: src/components/App/App.css
Comment (Sykogst) [inline src/components/App/App.css:21]: I really like this home button hover effect! Definitely think we should keep it.

I almost like it so much, I think it would be neat to somehow implement on the nav bar buttons for it to be neon highlighted when the user is on that current page, but might be toooo much.

Comment (ericahagle) [reply src/components/App/App.css:21]: @Sykogst I'm glad you like it! I certainly could add something like it to the buttons. I just wasn't sure if that would be too much or not. And I was kinda thinking of it like the app title was a "neon sign" in a way. Maybe this is one we should discuss as a group and then I can record whatever choice we make here in the PR along with changes (if any).

File: src/components/Navigation/Navigation.css
Comment (ericahagle) [reply src/components/Navigation/Navigation.css:11]: @Sykogst I was thinking the same. I'll try another font for the buttons before we call this PR done. We also have a ticket for finalizing overall app styling that we'll hit at the end of the project. So, if anyone has fonts that they prefer and want to suggest them, we can always switch 'em up later, too.

Comment (ericahagle) [reply src/components/Navigation/Navigation.css]: Updated the buttons font to a more simple, readable one.

Comment (tenthwalker) [reply src/components/Navigation/Navigation.css]: I like this font, does it come in multiple weights? A boldface version of it might make the buttons that much more legible.

File: src/components/LogoutButton/LogoutButton.css
Comment (tenthwalker) [inline src/components/LogoutButton/LogoutButton.css:1]: Was this commented out as in you opted to go a different route or as in it need to be added back later?

File: src/components/EncounterBuilderButton/EncounterBuilderButton.jsx
Comment (tenthwalker) [inline src/components/EncounterBuilderButton/EncounterBuilderButton.jsx:3]: Thanks for catching this, my mistake!

File: src/components/App/App.css
Comment (tenthwalker) [reply src/components/App/App.css:21]: I like the neon sign idea, we can play with that as we go along cuz I think especially as a hover state it's effective for navigation clarity.

**Review by tenthwalker — APPROVED** [review]
Looks good overall to me, thanks for updating the necessary imports and getting that responsive design dialed in!

File: src/components/App/App.css
Comment (ericahagle) [reply src/components/App/App.css:21]: Personally, I think it's ideal for one thing to stand out as special. But we can revisit using some level of it elsewhere, too. Maybe we can hold off on adding more of it until our final styling ticket.

File: src/components/EncounterBuilderButton/EncounterBuilderButton.jsx
Comment (ericahagle) [reply src/components/EncounterBuilderButton/EncounterBuilderButton.jsx:3]: Eh, I wouldn't even call it a mistake. We just probably didn't really need to add all those imports in until we're using them, since ya never know if it's gonna change when you start to implement. I'm sure there will be others that we'll just end up removing because we don't even use them. Nbd either way! 🙂

File: src/components/LogoutButton/LogoutButton.css
Comment (ericahagle) [reply src/components/LogoutButton/LogoutButton.css:1]: Ah yes, decided to do something different and just didn't delete. I'll clean that up before we merge.

File: src/components/Navigation/Navigation.css
Comment (ericahagle) [reply src/components/Navigation/Navigation.css]: @tenthwalker Good callout! Upped the weight to 700.
Screenshot 2024-02-05 at 1 21 00 PM

**Pull request merged**
